llvm/test/CodeGen/AArch64/hadd-combine.ll

```diff
@@ -341,6 +341,18 @@ define <8 x i16> @sub_fixedwidth_v4i32(<8 x i16> %a0, <8 x i16> %a1)  {
   ret <8 x i16> %res
 }
 
+define <8 x i16> @shsub_fixedwidth_v4i32(<8 x i16> %a0, <8 x i16> %a1)  {
```

Comment:
shsub_fixedwidth_v4i32 -> srhadd_fixedwidth_v8i16 (its matching the shsub instruction for v8i16 types)